### DISCUSSION TITLE
Does nanoid have async/await option or will it always block ?

### DISCUSSION THREAD

**Discussion by rajathgunners** [ai/nanoid#498] [General]
I just had a doubt on whether nanoid library can be used with await to make sure it does not block the main thread.
Does this code snippet work or have any effect?
`const id = await nanoid()`

**Comment by ai** [comment]
We had async version before, but not anymore.

The main problems are: 

1. It is slover
2. There is no async API in browser and React Native APIs
3. You will have benefits from async version only first call. Random generator use true randomness only at the beginning to hehe seed. Next it can respond immediately.
